Core/src/Exception/ServiceException.php

```diff
@@ -81,4 +83,27 @@ public function getMetadata()
     {
         return $this->metadata;
     }
+
+    /**
+     * Returns the ErrorInfo part of the exception
+     *
+     * @return array
+     */
+    public function getErrorInfo()
```

Comment:
If the exception bubbles up from the GAPIC/GAX layer, would we be able to utilize https://github.com/googleapis/gax-php/blob/main/src/ApiException.php#L116-L119?

Maybe we only need to perform this logic in the case that we use the handwritten's transport layer?